test/src/main/java/org/apache/accumulo/test/RecoveryIT.java

```diff
@@ -0,0 +1,226 @@
+
+      // create 3 flush files
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation("a");
+        m.put("b", "c", new Value("v"));
+        for (int i = 0; i < 3; i++) {
+          bw.addMutation(m);
+          bw.flush();
+          c.tableOperations().flush(tableName, null, null, true);
+        }
+        // create an unsaved mutation
+        bw.addMutation(m);
+      }
+
+      // Confirm that there are no walog entries for this table
+      assertEquals(0, countWaLogEntries(c, tid));
+
+      MiniAccumuloClusterImpl mini = (MiniAccumuloClusterImpl) cluster;
+      MiniAccumuloClusterControl control = (MiniAccumuloClusterControl) cluster.getClusterControl();
+
+      // Stop any running Compactors and ScanServers
+      control.stopAllServers(ServerType.COMPACTOR);
+      control.stopAllServers(ServerType.SCAN_SERVER);
+
+      // Kill the TabletServer in resource group that is hosting the table
+      List<Process> procs = control.getTabletServers(RESOURCE_GROUP);
+      assertEquals(1, procs.size());
+      procs.get(0).destroyForcibly().waitFor();
+      control.getTabletServers(RESOURCE_GROUP).clear();
+
+      // The TabletGroupWatcher in the Manager will notice that the TabletServer is dead
+      // and will assign the TabletServer's walog to all of the tablets that were assigned
+      // to that server. Confirm that walog entries exist for this tablet
+      Wait.waitFor(() -> countWaLogEntries(c, tid) > 0, 60_000);
+
+      // Start the process that will do the log sorting
+      switch (serverForSorting) {
+        case "TSERVER":
+          // We don't need to re-start the TabletServers here, there is
+          // already a TabletServer running in the default group that
+          // is hosting the root and metadata tables. It should perform
+          // the log sorting.
+          break;
+        case "SSERVER":
+          mini.getConfig().getClusterServerConfiguration().setNumDefaultScanServers(1);
+          getClusterControl().startAllServers(ServerType.SCAN_SERVER);
+          break;
+        case "COMPACTOR":
+          mini.getConfig().getClusterServerConfiguration().setNumDefaultCompactors(1);
+          getClusterControl().startAllServers(ServerType.COMPACTOR);
+          break;
+        case "ALL":
+        default:
+          fail("Unhandled server type: " + serverForSorting);
+      }
+
+      // Confirm sorting completed
+      Wait.waitFor(() -> logSortingCompleted(c, tid) == true, 60_000);
+
+      // Start the tablet servers so that the Manager
+      // can assign the table and so that recovery can be completed.
+      getClusterControl().startAllServers(ServerType.TABLET_SERVER);
+
+      c.instanceOperations().waitForBalance();
+
+      // When the tablet is hosted, the sorted walogs will be applied
+      // Confirm the 3 walog entries are gone for this tablet
+      assertEquals(0, countWaLogEntries(c, tid));
+
```

Comment:
Would be good to scan the tablet here to see if the data written by the batch writer is present.